javaScript-week2/candy-helper.js

```diff
@@ -0,0 +1,36 @@
+const boughtCandyPrices = [];
+
+const candyPrices = {
+  "sweet": 0.5,
+  "chocolate": 0.7,
+  "toffee": 1.1,
+  "chewing-gum": 0.03,
+};
+
+function addCandy(candyType, weight) {
+  const totalCost = candyPrices[candyType] * weight;
+  boughtCandyPrices.push(totalCost);
+}
+
+let amountToSpend = Math.random() * 100;
+
+function canBuyMoreCandy() {
+  let totalSpent = 0;
+  for (let i = 0; i < boughtCandyPrices.length; i++) {
+    totalSpent += boughtCandyPrices[i];
+  }
+
+  if (totalSpent < amountToSpend) {
+    console.log("You can buy more, so please do!");
+    return true;
+  } else {
+    console.log("Enough candy for you!");
+    return false;
+  }
+}
+
+addCandy("sweet", 10);
+addCandy("chocolate", 15);
+addCandy("", 10);
```

Comment:
Good check, but remember to add the logic to the function to handle this edge case. It's the same when the type doesn't match any of the names from the array up.